src/nomad_simulation_parsers/parsers/h5md/parser.py

```diff
@@ -239,63 +238,239 @@ def get_contributions(
+                if data_type in type_filter:
+                    result_dict = {'label': label}
+                    # Process data with consistent logic
+                    self._process_observable_data(
+                        result_dict, data_dict, data_type, type_filter
+                    )
+                    results.append(result_dict)
+        return results
+
+    def _process_observable_data(
+        self, result_dict: dict, data_dict: dict, data_type: str, type_filter: list
+    ) -> None:
+        """Helper method to consistently process observable data."""
+        for key in data_dict.keys():
+            if key != '@type':  # Skip type metadata
+                value = self.get_value(key, data_dict)
+                if value is not None:
+                    # Handle 'times' field only for correlation functions
+                    if key == 'times' and data_type != 'correlation_function':
+                        continue
+                    self._add_value_to_result(result_dict, key, value)
+
+    def _add_value_to_result(self, result_dict: dict, key: str, value) -> None:
+        """Helper method to add a value to result dict with proper unit handling."""
+        # Handle 'times' field as normal Quantity
+        if key == 'times':
+            result_dict[key] = value
+        # Handle 'value' field specially - always use schema-mapped field names
+        elif key == 'value':
+            if hasattr(value, 'magnitude') and hasattr(value, 'units'):
+                # Quantity with units
+                magnitude = value.magnitude
+                # Convert scalars to 1-element arrays for schema consistency
+                if not hasattr(magnitude, '__len__') or isinstance(magnitude, str):
+                    magnitude = [magnitude]
+                result_dict['value_magnitude'] = magnitude
+                result_dict['value_unit'] = str(value.units)
+            else:
+                # Raw value (list, float, etc.) - still use mapped field name
+                # Convert scalars to 1-element arrays for schema consistency
+                if not hasattr(value, '__len__') or isinstance(value, str):
+                    value = [value]
+                result_dict['value_magnitude'] = value
+        # Handle other Quantities by separating magnitude and unit
+        elif hasattr(value, 'magnitude') and hasattr(value, 'units'):
+            result_dict[key] = value.magnitude
+            result_dict[f'{key}_unit'] = str(value.units)
+        else:
+            result_dict[key] = value
+
+    def _get_observable_type(self, label: str, data_dict: dict) -> str | None:
+        """Get the type of an observable from H5MD file or parsed data."""
+        # Try to get type from raw H5MD file attributes
+        if hasattr(self, 'h5_parser') and hasattr(self.h5_parser, 'h5_archive'):
+            try:
+                h5_file = self.h5_parser.h5_archive
```

Comment:
I cannot seem to see where this is initialized but can you add `h5_parser` in l30 and provide type. I suggest you enclose this in a context 
```
with self.h5_parser.h5_archive as h5_file
```
or extend 
```
def close(self)
```

Reply:
I wasn't exactly sure what to do here, I'm not sure if `with self.h5_parser.h5_archive as h5_file` will work or how to do it properly...I just tried to add some context of where the file is being opened and closed...

Reply:
I guess I just don't understand what `self.h5_parser.h5_archive.close()` does, is this not closing the file? Why would we do that at the end of this function?

Reply:
we can discuss it further in discord.